File: Data-Structures/Queues/queue.cpp
# Python Implementation of a Queue using List

class Queue:
    def __init__(self):
        self.queue = []
    
    def enqueue(self, item):
        """Add an item to the queue."""
        self.queue.append(item)
        print(f"Enqueued: {item}")
    
    def dequeue(self):
        """Remove an item from the queue."""
        if self.is_empty():
            print("Queue is empty!")
            return None
        return self.queue.pop(0)
    
    def is_empty(self):
        """Check if the queue is empty."""
        return len(self.queue) == 0
    
    def front(self):
        """Get the front element of the queue."""
        if self.is_empty():
            print("Queue is empty!")
            return None
        return self.queue[0]
    
    def display(self):
        """Display the queue."""
        print("Queue:", self.queue)

# Example Usage
q = Queue()
q.enqueue(10)
q.enqueue(20)
q.enqueue(30)
q.display()
print("Dequeued:", q.dequeue())
q.display()

File: Data-Structures/Queues/queue.js
// JavaScript Implementation of a Queue using Array

class Queue {
    constructor() {
        this.queue = [];
    }

    enqueue(item) {
        this.queue.push(item);
        console.log(`Enqueued: ${item}`);
    }

    dequeue() {
        if (this.isEmpty()) {
            console.log("Queue is empty!");
            return null;
        }
        console.log(`Dequeued: ${this.queue[0]}`);
        return this.queue.shift();
    }

    isEmpty() {
        return this.queue.length === 0;
    }

    front() {
        if (this.isEmpty()) {
            console.log("Queue is empty!");
            return null;
        }
        console.log(`Front element: ${this.queue[0]}`);
    }

    display() {
        console.log("Queue:", this.queue);
    }
}

// Example Usage
const q = new Queue();
q.enqueue(10);
q.enqueue(20);
q.enqueue(30);
q.display();
q.dequeue();
q.display();
